server/core/include/irods/configuration_hook_manager.hpp

```diff
@@ -0,0 +1,65 @@
+#ifndef CONFIGURATION_HOOK_MANAGER_HPP
+#define CONFIGURATION_HOOK_MANAGER_HPP
+
+#include <vector>
+#include <string>
+#include <nlohmann/json.hpp>
+/// \file
+
+namespace irods::experimental::hooks{
```

Comment:
Why is the `hooks` namespace needed?
What else would go in this namespace?

Reply:
I wasn't sure how much infrastructure it was going to need, so putting it in its own namespace seemed like not a bad first decision

Probably doesn't need that with the current code